enhancements/etcd/automated-backups.md

```diff
@@ -0,0 +1,284 @@
+        // The max should be smaller than the scheduled interval
+        // for the next backup else the alert will never be generated
+    MaxDurationToCompleteBackup string `json: "maxDurationWithoutBackup"`
+    
+    // RetentionCount defines the maximum number of backups to retain.
+    // If the number of successful backups matches retentionCount 
+    // the oldest backup will be removed before a new backup is initiated.
+    // The count here is for the total number of backups
+    // TODO: Validation on min and max counts.
+    // TODO: Retention could also be based on the total size, or time period
+        // e.g retain 500mb of backups, or discard backups older than a month
+        // If we want to extend this API in the future we should make retention
+        // a type so we can add more optional fields later
+    RetentionCount int `json: "retentionCount"`
+    
+}
+```
+
+**TODO:** How do we feature gate the entire type?
+For a field within a non-gated type it seems to be the tag [`// +openshift:enable:FeatureSets=TechPreviewNoUpgrade`](https://github.com/openshift/api/blob/master/example/v1/types_stable.go#L33). No such tag visible on the [whole type gated example](https://github.com/openshift/api/blob/1957a8d7445bf2332f027f93a24d7573f77a0dc0/example/v1alpha1/types_notstable.go#L19).
+
+
+```Go
+// EtcdBackupStatus represents the status of the EtcdBackup CRD.
+type EtcdBackupStatus struct {
+
+    // Conditions represents the observations of the EtcdBackup's current state.
+    // TODO: Identify different condition types/reasons that will be needed.
+    // +optional
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+    
+    // ObservedGeneration is the most recent generation observed for this
+    // EtcdBackup. It corresponds to the EtcdBackup's .metadata.generation that the condition was set based upon.
+    // This lets us know whether the latest condition is stale.
+    // +optional
+    ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+    // TODO: How do we track the state of the current/last backup?
+        // Would condition states be enough? e.g Created/In-progress/Complete/Failed/Unknown
+    // Track metadata on last successful backup?
+        // node name, path, size, timestamp
+    // With backup retention do we also track the last backup removed?
+}
+```
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Backup controller
+
+A backup controller in the cluster-etcd-operator will reconcile the CRD to execute the backup schedule and report the status of ongoing backups and failures to the `EtcdBackup` status.
+
+There are different options to explore on how we want to execute saving the backup snapshot and any other required metadata. As well as how we enforce the schedule.
+
+#### Execution method
+
+Create a backup pod that runs the backup script on a designated master node to save the backup file on the node's host filesystem:
```

Comment:
the job construct seems like a good choice here.  I don't object to a pod, but are you sure you don't want a job instead?